nflow-engine/src/main/java/io/nflow/engine/workflow/curated/SimpleState.java

```diff
@@ -0,0 +1,45 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.model.ModelObject;
+import io.nflow.engine.workflow.definition.WorkflowState;
+import io.nflow.engine.workflow.definition.WorkflowStateType;
+
+public class SimpleState extends ModelObject implements WorkflowState {
```

Comment:
Add some JavaDoc to describe what this curated class is? I suppose this is "the default implementation", which would be named `WorkflowState` if the interface had not stolen the name already? And `WorkflowStateImpl` sounds archaic.

Still, `SimpleState` makes me look for `ComplexState` or alternative state implementations. I expect customizing state implementation will be pretty rare. Naming is difficult, and I can go forward with `SimpleState` too, but right now e.g. `BaseState` would sound better - communicates a possible extension point and makes you less surprised there's no offered alternatives.

Reply:
Renamed `SimpleState` -> `State` and added javadoc

Reply:
Great! I approve :)